src/Controller/EnMarche/DonationController.php

```diff
@@ -133,4 +137,44 @@ public function resultAction(Request $request, Donation $donation)
             'retry_url' => $retryUrl,
         ]);
     }
+
+    /**
+     * @Route(
+     *     "/mensuel/cancel",
+     *     defaults={"_enable_campaign_silence"=true},
+     *     name="donation_subscription_cancel"
+     * )
+     * @Method("GET")
+     * // TODO: make test
+     */
+    public function cancelSubscriptionAction(Request $request, DonationRepository $donationRepository, PayboxPaymentUnsubscription $payboxPaymentUnsubscription): RedirectResponse
+    {
+        $donations = $donationRepository->findAllSubscribedDonationByEmail($this->getUser()->getEmailAddress());
+
+        foreach ($donations as $donation) {
+            try {
+                $payboxPaymentUnsubscription->unsubscribe($donation);
+                $this->getDoctrine()->getManager()->flush();
+                $payboxPaymentUnsubscription->sendConfirmationMessage($donation, $this->getUser());
+                $this->addFlash(
+                    'success',
+                    'Votre don mensuel a bien été annulé. Vous recevrez bientôt un mail de confirmation.'
+                );
+            } catch (PayboxPaymentUnsubscriptionException $payboxPaymentUnsubscriptionException) {
```

Comment:
`PayboxPaymentUnsubscriptionException $payboxPaymentUnsubscriptionException`
vs
`PayboxPaymentUnsubscriptionException $e`
IMO: the second one is better